snapcraft/remote/git.py

```diff
@@ -181,23 +184,66 @@ def push_url(
             "Pushing %r to remote %r with refspec %r.", ref, stripped_url, refspec
         )
 
-        # Create a list of tags to push
+        # temporarily call git directly due to libgit2 bug that unable to push
+        # large repos using https. See https://github.com/libgit2/libgit2/issues/6385
+        cmd: list[str] = ["git", "push", remote_url, refspec, "--progress"]
         if push_tags:
-            tag_refs = [
-                t.name
-                for t in self._repo.references.iterator(pygit2.GIT_REFERENCES_TAGS)
-            ]
-        else:
-            tag_refs = []
+            cmd.append("--tags")
+
+        git_proc: Optional[subprocess.Popen] = None
         try:
-            self._repo.remotes.create_anonymous(remote_url).push([refspec] + tag_refs)
-        except pygit2.GitError as error:
+            with subprocess.Popen(
+                cmd,
+                cwd=str(self.path),
+                bufsize=1,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            ) as git_proc:
+                # do not block on reading from the pipes (not working on Windows)
```

Comment:
Can you rephrase this?

I'm not sure if it means:
1. You added this code so that it does work on windows
2. You added this code and it does not work on windows

If you mean option 1, then something like `set non-blocking to prevent read blocking on windows` may be more clear

Reply:
I mean the `os.set_blocking()` is not supported in Windows until Python 3.12. So it will have exception or still have blocked I/O in Windows. Not sure if anyone actually use it.